src/oscar/templates/oscar/search/results.html

```diff
@@ -42,13 +42,12 @@
 
 {% block content %}
     <form method="get" action="." class="form-horizontal">
-        {# Render other search params in a hidden block #}
-        <div style="display:none">
-            {% for value in selected_facets %}
-                <input name="selected_facets" value="{{ value }}" />
-            {% endfor %}
-            {{ search_form.q }}
-        </div>
+        {# Render other search params as hidden inputs #}
+        {% for value in selected_facets %}
+            <input type="hidden" name="selected_facets" value="{{ value }}" />
+        {% endfor %}
+        <input type="hidden" name="q" value="{{ search_form.q.value }}" />
```

Comment:
Again, that inserts an ID attribute which is the whole issue this PR is trying to address. The options are:

- Render one of those forms manually.
- turn off auto_id on the search form class.

I am happy to do the latter, but it risks breaking a lot more things for people relying on that ID being there.